File: VB/MyGridRegistration.vb
Imports System
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Drawing
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.ViewInfo
Imports DevExpress.XtraGrid.Registrator

Namespace CustomGrid
	Public Class MyGridViewInfoRegistrator
		Inherits GridInfoRegistrator

		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyGridView"
			End Get
		End Property
		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New MyGridView(TryCast(grid, GridControl))
		End Function
		Public Overrides Function CreatePainter(ByVal view As BaseView) As BaseViewPainter
			Return New MyGridPainter(TryCast(view, MyGridView))
		End Function
	End Class
	Public Class MyBandedGridViewInfoRegistrator
		Inherits BandedGridInfoRegistrator

		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyBandedGridView"
			End Get
		End Property
		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New MyBandedGridView(TryCast(grid, GridControl))
		End Function
		Public Overrides Function CreatePainter(ByVal view As BaseView) As BaseViewPainter
			Return New MyBandedGridPainter(TryCast(view, MyBandedGridView))
		End Function
	End Class
End Namespace

File: CS/MyGridRegistration.cs
using System;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Drawing;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.ViewInfo;
using DevExpress.XtraGrid.Registrator;

namespace CustomGrid {
	public class MyGridViewInfoRegistrator : GridInfoRegistrator {
		public override string ViewName { get { return "MyGridView"; } }
		public override BaseView CreateView(GridControl grid) { return new MyGridView(grid as GridControl); }
		public override BaseViewPainter CreatePainter(BaseView view) { return new MyGridPainter(view as MyGridView); }
	}
	public class MyBandedGridViewInfoRegistrator : BandedGridInfoRegistrator {
		public override string ViewName { get { return "MyBandedGridView"; } }
		public override BaseView CreateView(GridControl grid) { return new MyBandedGridView(grid as GridControl); }
		public override BaseViewPainter CreatePainter(BaseView view) { return new MyBandedGridPainter(view as MyBandedGridView); }
	}
}
